exchange/utils.go

```diff
@@ -177,48 +172,79 @@ func (rs *requestSplitter) cleanOpenRTBRequests(ctx context.Context,
 			}
 		}
 
+		ipConf := privacy.IPConf{IPV6: auctionReq.Account.Privacy.IPv6Config, IPV4: auctionReq.Account.Privacy.IPv4Config}
+
+		reqWrapper := cloneBidderReq(bidderRequest.BidRequest)
+
+		// FPD should be applied before policies, otherwise it overrides policies and activities restricted data
+		if auctionReq.FirstPartyData != nil && auctionReq.FirstPartyData[bidderRequest.BidderName] != nil {
+			applyFPD(auctionReq.FirstPartyData[bidderRequest.BidderName], reqWrapper)
+		}
+
 		passIDActivityAllowed := auctionReq.Activities.Allow(privacy.ActivityTransmitUserFPD, scopedName, privacy.NewRequestFromBidRequest(*req))
 		if !passIDActivityAllowed {
-			privacyEnforcement.UFPD = true
+			//UFPD
+			privacy.ScrubDeviceIDs(reqWrapper)
+			privacy.ScrubUserIDs(reqWrapper)
+			privacy.ScrubUserExt(reqWrapper, "data")
```

Comment:
As a general approach, I recommend calling just one scrubber method and manage the groupings in the scrubber code. This could be `ScrubUserFPD`, which [per the spec](https://docs.google.com/document/d/1dRxFUFmhh2jGanzGZvfkK_6jtHPpHXWD7Qsi6KEugeE/edit#heading=h.rwnqutpvmq) includes:

``` 
user.eids, user.ext.data.*, user.data.*, user.{id, buyeruid, yob, gender} and device-specific IDs
```

Looks like we're missing at least eids.

Reply:
Good catch! Refactored

Reply:
I'll add more unit tests for these cases